### ISSUE TITLE
Don't diff files that don't actually differ

### ISSUE THREAD

**Issue by rogeruiz** [rogeruiz/.files#13]
If files aren't actually different, then don't `opendiff`/`vimdiff` them.

**Comment by rogeruiz** [comment]
For anyone interested in helping out with this, I'll gladly walk through the thought process and pair on this. 👍

**Comment by christopherwood** [comment]
I found the link to this retweeted in my twitter timeline. My thoughts are that you can pre-diff them or compare checksums.

First POC, pre-diff if you have GNU diff or a similar diff installed (copied from my command line, needs something to check for diff):

```
echo aaa >/tmp/one
echo bbb >/tmp/two
[[ "0" == "`diff /tmp/one /tmp/two | wc -l`" ]] || echo opendiff them
```
Second POC, find some checksum program and use that.

```
#!/bin/bash

# http://redsymbol.net/articles/unofficial-bash-strict-mode/
set -euo pipefail

filea=/tmp/a
fileb=/tmp/b
echo 123 >$filea
echo 123 >$fileb
# deliberately blank
sum=
[[ -x "`which md5sum`" ]] && sum=md5sum; export sum
[[ -x "`which sha1sum`" ]] && sum=sha1sum; export sum
[[ -x "`which sha256sum`" ]] && sum=sha256sum; export sum
suma=`${sum} ${filea} | cut -f 1 -d " "`
sumb=`${sum} ${fileb} | cut -f 1 -d " "`
[[ "${suma}" == "${sumb}" ]] || echo opendiff things here
```

Obviously both quick-and-dirty but you get the idea.

(If you have a local perl you might think of GNU stow, but you appear to be on a mac and I don't know if perl ships there. https://www.gnu.org/software/stow/)

**Comment by christopherwood** [comment]
One more thought. Yes, diff in theory exits 1 for different files and 0 for identical files, however relying on the same behaviour from the same utilities across unixes is the path to pain. Best to keep it less implicit and actually count the lines.

**Comment by rogeruiz** [comment]
Closed by df3f86735c882ba28e66d26096c341379be5e799
